.github/workflows/tests.yml

```diff
@@ -43,17 +43,18 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+      - name: Read charmcraft version file
+        id: charmcraft
+        run: echo "channel=$(cat .charmcraft-channel)" >> $GITHUB_OUTPUT        
       - name: Setup Python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
           python-version: '3.10'
-      - name: Install Dependencies
-        run: |
-          pip install tox
       - name: Setup operator environment
         uses: charmed-kubernetes/actions-operator@main
         with:
-          juju-channel: 3.1/stable
+          juju-channel: 3/stable
```

Comment:
juju 3 does not support bionic so that's probably why the tests fail (no idea why they didn't previously though)

Reply:
it looks like bionic didn't come alive b/c of the version of python didn't mesh well with the newer version of ops.  I'm backing up to `ops<2.0` to see how well that goes.